cpp/src/arrow/type_traits.h

```diff
@@ -639,10 +681,9 @@ template <typename T, typename R = void>
 using enable_if_fixed_width_type = enable_if_t<is_fixed_width_type<T>::value, R>;
 
 template <typename T>
-using is_binary_like_type =
-    std::integral_constant<bool, (is_base_binary_type<T>::value &&
-                                  !is_string_like_type<T>::value) ||
-                                     is_fixed_size_binary_type<T>::value>;
+using is_binary_like_type = std::integral_constant<
+    bool, (is_base_binary_type<T>::value && !is_string_like_type<T>::value) ||
+              is_binary_view_type<T>::value || is_fixed_size_binary_type<T>::value>;
```

Comment:
This changes the meaning of `is_binary_like_type`. For list-views I added new predicates and kept the existing predicates' semantics.

Illustrate in this graph https://gist.github.com/felipecrv/3c02f3784221d946dec1b031c6d400db

Reply:
I agree with @felipecrv btw. This may silently produce incorrect binary-view handling in operations not touched by this PR.

Reply:
These traits are used in very few places. I think the current arrangement is fine; "binary like" = has a byte array, "string like" = has a byte array which is utf8.

Reply:
I've made a follow up to address this https://github.com/apache/arrow/issues/38204

Reply:
I've reverted the change to `is_binary_like_type`  and `is_string_like_type`